### PR TITLE
Change relative import to extensioned variations for Node16 and NodeNext

### LINKED ISSUE #6377: Types are broken for moduleResolution: Node16 and NodeNext
### Core Library

MSAL Node (@azure/msal-node)

### Core Library Version

2.0.2

### Wrapper Library

Not Applicable

### Wrapper Library Version

None

### Public or Confidential Client?

Public, Confidential

### Description

reminiscent of #6151 

I'm using latest typescript, my server is fully esm.
```json
{
  "type": "module"
}
```
```json
{
  "compilerOptions": {
    "module": "NodeNext",
    "moduleResolution": "NodeNext"
  }
}
```
and all the types from msal are broken. This is because typescript doesn't map `.mjs` to `.d.ts` naturally. So you get this, if you're looking inside:
image

Looking at `index.d.ts`, for `esm` the imports don't point to any _typed_ file,
image

If directed explicitly to the `d.ts`, this is resolved

image

### **!NOTE! This is not enough, for this to work _all_  imports in `d.ts` files have to explicitly point to `d.ts` files**

I'm not sure how to solve this, but from my personal experience I found that meme extensions are very painful to work with

### Error Message

_No response_

### Msal Logs

_No response_

### MSAL Configuration

```javascript
{}
```

### Relevant Code Snippets

```javascript
import { CryptoProvider } from "@azure/msal-node"
// or anything, really
```

### Reproduction Steps

1. create modern esm node project
2. install `@azure/msal-node`
3. ??????
4. profit

### Expected Behavior

Typed correctly

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

None (Server)

### Regression

@azure/msal-node@~2

### Source

External (Customer)

### PR DESCRIPTION
**Pull request by tnorling**
Addresses Node 16 and NodeNext moduleResolution types

Thanks @datner for the PR!

### CODE REVIEW

**Comment by datner** [comment]
🎉

File: change/@azure-msal-node-fd9abab4-f755-4c59-8e63-744eefaee29b.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat(msal-node): change all relative imports to .js",
```
Comment (konstantin-msft) [inline change/@azure-msal-node-fd9abab4-f755-4c59-8e63-744eefaee29b.json]: change #?

**Review by konstantin-msft — APPROVED** [review]

File: change/@azure-msal-node-fd9abab4-f755-4c59-8e63-744eefaee29b.json (hunk 1 above)
Comment (datner) [reply change/@azure-msal-node-fd9abab4-f755-4c59-8e63-744eefaee29b.json]: I don't understand 😄

Comment (tnorling) [inline change/@azure-msal-node-fd9abab4-f755-4c59-8e63-744eefaee29b.json]:
```suggestion
  "comment": "feat(msal-node): change all relative imports to .js #6399",
```

**Review by sameerag — APPROVED** [review]

**Pull request merged**